qiskit/circuit/qpy_serialization.py

```diff
@@ -367,6 +411,22 @@ class if it's defined in Qiskit. Otherwise it falls back to the custom
 FILE_HEADER_SIZE = struct.calcsize(FILE_HEADER_PACK)
 
 # HEADER binary format
+HEADER_V2 = namedtuple(
+    "HEADER",
+    [
+        "name_size",
+        "global_phase_type",
+        "global_phase_size",
+        "num_qubits",
+        "num_clbits",
+        "metadata_size",
+        "num_registers",
+        "num_instructions",
+    ],
+)
+HEADER_V2_PACK = "!H1cHIIQIQ"
```

Comment:
Does the explicit `1c` in the string have different semantics to just `c`?

Reply:
No they should be the same, `1c` means just 1 repeat of `c`. It was just a habit from when I was using fixed length strings in the first draft of QPY. It makes more sense if you do something like `4c` which is the same as `cccc`. It's documented here: https://docs.python.org/3/library/struct.html#format-characters (it's the first paragraph after the numbered `Notes` list). I can change it to just be `c` if you'd prefer that.

Reply:
I have no preference, I was just wondering if a count of one might have implied something else.  The idea of fixed-length C strings seems totally valid to me, and I'm happy the way it is.